### PR TITLE
texlive: 2018 -> 2018-final

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

It's March, final texlive is out.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @bhipple

### CODE REVIEW

**Review by xeji — APPROVED** [review]
Thank you. Don't have time to test this locally but LGTM. Should be backported to 19.03.

File: pkgs/tools/typesetting/tex/texlive/default.nix [hunk 1]
```diff
@@ -108,23 +108,7 @@ let
...
-        https://ipfs.io/ipfs/QmT4Z67wXin1Z9DhvqwSSkSZSuu8hT6LgDyMu6CBm9Tb7t/tlnet/archive
-
-        # The canonical source moves quickly and will be broken almost immediately
-        http://mirror.ctan.org/tex-archive/systems/texlive/tlnet/archive
-
-        # Should be stable for historic, archived releases
-        # http://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2018/tlnet-final/archive
-        # TODO: use this later when 2018 is archived
-
+        http://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2018/tlnet-final/archive
```
Comment (dotlambda) [inline pkgs/tools/typesetting/tex/texlive/default.nix]: If multiple mirros are desirable you could add `ftp://tug.org/texlive/historic/2018/tlnet-final/archive/`.

**Comment by dotlambda** [comment]
Biber needed an update because according to https://sourceforge.net/projects/biblatex-biber/
> Biber 2.12 is for biblatex 3.12

**Comment by dotlambda** [comment]
Maybe we should add a comment somewhere to remind people to update Biber whenever biblatex is updated.

**Comment by veprbl** [comment]
@dotlambda
How about this for solution?

**Comment by dotlambda** [comment]
> How about this for solution?

Perfect!

Should we disable Biber's tests on ARM? I have no idea how to solve
> Can't load '/nix/store/0wdrvy0yyqzjf20q2m2fw8njbk0c1367-perl5.28.1-Text-BibTeX-0.85/lib/perl5/site_perl/5.28.1/aarch64-linux-thread-multi/auto/Text/BibTeX/BibTeX.so' for module Text::BibTeX: libbtparse.so: cannot open shared object file: No such file or directory at /nix/store/sg676sz2hbjd3vhiq5zgyndbzka59iq1-perl-5.28.1/lib/perl5/5.28.1/aarch64-linux-thread-multi/DynaLoader.pm line 193.

**Comment by veprbl** [comment]
> Should we disable Biber's tests on ARM?
I would imagine biber needs a working TextBibTeX to operate, so this would probably just break at runtime.

**Review by dotlambda — APPROVED** [review]
I think this is ready to be merged. Tested locally and everything works fine.

**Comment by xeji** [comment]
backported to 19.03 in  9e762daa738..734fdd9916c

**Comment by vcunat** [comment]
Perhaps we should change the `maintainer` field to something closer to the reality?  I'm afraid I won't be putting much effort towards TeX stuff anymore (and I wasn't for the past couple of years already).

**Pull request merged**